### PR TITLE
Dropdown layout footer header

### PR DESCRIPTION
**Pull request by lbelinsk**
1. Fixes from the previous branch (without MultiSelect).
2. DropdownLayout fixedFooter + fixedHeader.
3. Migration of InputWithOptions to use the testkit template.
4. Migration of DropdownLayout to use the testkit template.
5. Migration of AutoComplete to use the testkit template.
6. Migration of Dropdown to use the testkit template.
7. Migration of Tag to use the testkit template.
8. Added MultiSelect new version with testkit.
9. Changed the usage of Tag. instead of ```<Tag label="something"/>```  use  ```<Tag>something</Tag>```
10. Docs update.

### CODE REVIEW

File: src/DropdownLayout/DropdownLayout.driver.js [hunk 1]
```diff
@@ -2,24 +2,26 @@ import React from 'react';
 import DropdownLayout from '../DropdownLayout';
 import ReactTestUtils from 'react-addons-test-utils';
 import ReactDOM from 'react-dom';
-import $ from 'jquery';
+import find from 'lodash.find';
```
Comment (mastertheblaster) [inline src/DropdownLayout/DropdownLayout.driver.js]: Perhaps not needed?

```js
[1, 2, 3].find(i => i === 2)
```

Does the same job

Comment (lbelinsk) [inline src/DropdownLayout/DropdownLayout.driver.js]: I did a mistake while merging.. I will take a look in a second

File: src/DropdownLayout/DropdownLayout.scss [hunk 2]
```diff
@@ -1,124 +1,137 @@
 @import "../common.scss";
 
 $option_height: 35px;
+$content_container_height: 260px;
 
 :export {
-    option_height: $option_height;
+  option_height: $option_height;
+  content_container_height: $content_container_height;
```
Comment (mastertheblaster) [inline src/DropdownLayout/DropdownLayout.scss]: @argshook I guess this is how you can export some variables? and use in a 3rd party project.

File: src/DropdownLayout/DropdownLayout.scss [hunk 3]
```diff
@@ -1,124 +1,137 @@
...
+  opacity: 0;
+  overflow: auto;
+  padding: 0;
+  transition: opacity 0.2s ease;
+  visibility: hidden;
+  width: 100%;
+  z-index: 6;
+  position:absolute;
+
+  -webkit-touch-callout: none;
```
Comment (mastertheblaster) [inline src/DropdownLayout/DropdownLayout.scss]: Guess you don't need them. This should be handled by css prefixer.

File: src/Input/Input.js [hunk 4]
```diff
@@ -28,6 +25,10 @@ class Input extends WixComponent {
 
   state = {
     focus: false
+  };
+
+  componentDidMount() {
+    this.props.autoFocus && this._onFocus();
```
Comment (mastertheblaster) [inline src/Input/Input.js]: Shouldn't this be like:
```js
this.props.autoFocus && this.focus();
```

Comment (nirhart) [inline src/Input/Input.js]: this.focus() don't call onFocus prop, we need that call so the user will know to show the dropdownLayout if exists

File: src/DropdownLayout/DropdownLayout.driver.js (hunk 1 above)
Comment (lbelinsk) [inline src/DropdownLayout/DropdownLayout.driver.js]: this wouldn't work since surprisingly options.childNodes is not an array. it is an object

Comment (lbelinsk) [inline src/DropdownLayout/DropdownLayout.driver.js]: I did console.log(typeof options.childNodes) and it showed object

**Pull request merged**